File: frontend/src/api/ai.ts
import { client } from "./client";

export interface TaxonomyDetails {
    kingdom?: string;
    division?: string;
    class_name?: string;
    order?: string;
    family?: string;
    genus?: string;
    species?: string;
}

export interface PlantAIDetailsResponse {
    common_name?: string;
    category?: string;
    planting_place?: "Indoor" | "Outdoor" | "Indoor & Outdoor";
    description?: string;
    common_diseases?: string;
    care_data?: Record<string, string>;
    taxonomy?: TaxonomyDetails;
    icon_url?: string;
    image_url?: string;
}

export interface PlantImageResponse {
    icon_url?: string;
    image_url?: string;
    page: number;
}

export const aiApi = {
    generatePlantDetails: async ({ commonName, scientificName }: { commonName?: string, scientificName?: string }): Promise<PlantAIDetailsResponse> => {
        const response = await client.post<PlantAIDetailsResponse>("/ai/generate-plant-details", {
            common_name: commonName,
            scientific_name: scientificName
        });
        return response.data;
    },
    fetchPlantImages: async ({ plantName, page }: { plantName: string, page?: number }): Promise<PlantImageResponse> => {
        const response = await client.get<PlantImageResponse>("/ai/fetch-plant-images", {
            params: { plant_name: plantName, page: page || 1 }
        });
        return response.data;
    }
};

File: backend/app/schemas/ai.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel

class PlantAIDetailsRequest(BaseModel):
    common_name: str

class TaxonomyDetails(BaseModel):
    kingdom: Optional[str] = None
    division: Optional[str] = None
    class_name: Optional[str] = None  # Using class_name since class is reserved
    order: Optional[str] = None
    family: Optional[str] = None
    genus: Optional[str] = None
    species: Optional[str] = None

class PlantAIDetailsResponse(BaseModel):
    category: Optional[str] = None
    planting_place: Optional[str] = None  # Indoor, Outdoor, or "Indoor & Outdoor"
    description: Optional[str] = None
    care_data: Optional[Dict[str, Any]] = None
    taxonomy: Optional[TaxonomyDetails] = None
